src/scala/co/actioniq/luna/dao/IgnoreUpdateCompiler.scala

```diff
@@ -0,0 +1,111 @@
+package co.actioniq.luna.dao
+
+import slick.compiler.InsertCompiler.Mode
+import slick.compiler.{CompilerState, Phase}
+import slick.util.ConstArray
+
+/**
+  * The goal of this compile stage is to remove fields from update statements that should not be updated when using
+  * case classes.  This is useful for updating using a case class and ignoring indexed fields, which would cause a
+  * deadlock.  This only removes fields from updating using a case class, sqlu and updating specific fields
+  * are not affected by this compiler.
+  *
+  * How it works:
+  * 1. Assumes ResultSetMapping state (probably after hoistClientOps).
+  * 2. Looks for fields in "from" with column option of IgnoreFieldForUpdate.  Removes those.
+  * 3. Removes from mapper fields with same ordinal position as fields in #2.
+  * 4. Creates a new "wrapper" function around mapper (mapper function goes from case class to tuple)
+  * that will create a tuple without ignored fields.
+  */
+class IgnoreUpdateCompiler extends Phase {
+  import slick.ast._
+  override val name: String = "ignoreUpdateCompiler"
+
+  override def apply(state: CompilerState): CompilerState = {
+    var fieldsToRemove: Map[Int, Boolean] = Map()
+    state.map {
+      case rsm @ ResultSetMapping(_, from, map) =>
+        val newFrom = from match {
+          case b @ Bind(sym, _, bindSelect) =>
+            val newSelect = bindSelect match {
+              case p @ Pure(pureSelect, _) =>
+                val filteredSelect = pureSelect match {
+                  case struct @ StructNode(elements) if elements.forall{ case (_, Select(Ref(str), _)) if str == sym => true; case _ => false} =>
+                    val fields = elements.map {
+                      case (_, s @ Select(Ref(_), fieldInfo: FieldSymbol)) => fieldInfo
+                    }
+                    fieldsToRemove = getFieldsToRemove(fields)
+                    struct.copy(elements=filterImportantFields(elements))
+                  case n => n
+                }
+                p.copy(value=filteredSelect)
+              case n => n
+            }
+            b.copy(select = newSelect).infer()
+          case n => n
+        }
+        val newMap = map match {
+          case typey @ TypeMapping(child, mapper, _) =>
+            val newChild = child match {
+              case pn @ ProductNode(children) =>
+                val newChildren = children.toSeq.zipWithIndex.filter { row =>
+                  !fieldsToRemove(row._2)
+                }.map(_._1).zipWithIndex.map(_._1)
```

Comment:
same comment here about collect instead of filtermap

Reply:
also unnecessary zipWithIndex?

Reply:
why did i zip with index....